chat-plugins/info.js

```diff
@@ -1833,145 +1833,95 @@ var commands = exports.commands = {
-				gameType: format.gameType || 'singles',
-				mod: format.mod,
-				searchShow: format.searchShow,
-				desc: format.desc || 'No description.'
-			});
+		if (!target) {
+			return this.sendReplyBox(
+				"- <a href=\"https://www.smogon.com/tiers/\">Smogon Tiers</a><br />" +
+				"- <a href=\"https://www.smogon.com/forums/threads/3498332/\">Tiering FAQ</a><br />" +
+				"- <a href=\"https://www.smogon.com/xyhub/tiers\">The banlists for each tier</a><br />" +
+				"<br /><em>Type /formatshelp <strong>[format|section]</strong> to get details about an available format or group of formats.</em>"
+			);
 		}
-		this.sendReplyBox(
-			"Available Formats: (<strong>Bold</strong> formats are on ladder.)<br />" +
-			buf.map(function (data) {
-				var str = "";
-				// Bold = Ladderable.
-				str += (data.searchShow ? "<strong>" + data.name + "</strong>" : data.name) + ": ";
-				str += "(" + (!data.mod || data.mod === 'base' ? "" : data.mod + " ") + data.gameType + " format) ";
-				str += data.desc;
-				return str;
-			}).join("<br />")
-		);
-	},*/
+		var targetId = toId(target);
+		if (targetId === 'ladder') targetId = 'search';
+		if (targetId === 'all') targetId = '';
+
+		var formatList;
+		var format = Tools.getFormat(format);
+		if (format.effectType === 'Format') formatList = [format.id];
+		if (!formatList) {
+			if (this.broadcasting && (room.id === 'lobby' || room.battle)) return this.sendReply("This command is too spammy to broadcast in lobby/battles");
+			formatList = Object.keys(Tools.data.Formats).filter(function (formatid) {return Tools.data.Formats[formatid].effectType === 'Format';});
+		}
+
+		// Filter formats and group by section
+		var exactMatch = '';
+		var sections = {};
+		var totalMatches = 0;
+		for (var i = 0; i < formatList.length; i++) {
+			var format = Tools.getFormat(formatList[i]);
+			var sectionId = toId(format.section);
+			if (targetId && !format[targetId + 'Show'] && sectionId !== targetId && !format.id.startsWith(targetId)) continue;
+			totalMatches++;
+			if (!sections[sectionId]) sections[sectionId] = {name: format.section, formats: []};
+			sections[sectionId].formats.push(format.id);
+			if (format.id !== targetId) continue;
+			exactMatch = sectionId;
+			break;
+		}
+
+		if (!totalMatches) return this.sendReply("No " + (target ? "matched " : "") + "formats found.");
+		if (totalMatches === 1) {
+			var format = Tools.getFormat(Object.values(sections)[0].formats[0]);
+			return this.sendReplyBox(format.desc.join("<br />"));
+		}
+
+		// Build tables
+		var buf = [];
+		for (var sectionId in sections) {
+			if (exactMatch && sectionId !== exactMatch) continue;
+			buf.push("<h3>" + Tools.escapeHTML(sections[sectionId].name) + "</h3>");
+			buf.push("<table style=\"border:1px solid gray\" cellspacing =\"0\" cellpadding=\"5\"><thead><th style=\"border:1px solid gray\" >Name</th><th style=\"border:1px solid gray\" >Mode</th><th style=\"border:1px solid gray\" >Description</th></thead><tbody>");
```

Comment:
Blah, you're going to need to set some sort of border-collapse option if you want this to work correctly...